### ISSUE TITLE
Create bucket (volume) failed

### ISSUE THREAD

**Issue by wsj0000** [chubaofs/chubaofs#869]
Errors got when I create bucket (also named volume in cfs). 
The aws-cli returns 'The bucket does not exist'.
![image](https://user-images.githubusercontent.com/11285030/91846395-c309d700-ec8c-11ea-8db2-37739b05b6fe.png)
And then I tried 'createVol' RESTful api, and the bucket create successfully.
![image](https://user-images.githubusercontent.com/11285030/91847319-fea4a100-ec8c-11ea-96da-442479eb09ee.png)
And 'ListBucket' is ok.
So I went to checking the source code. I saw it already been support.
And tried to create 'test1', still got the same error.
Then I checked the log. Found the log below
```
2020/09/01 19:37:21.096488 [DEBUG] auth_signature_v4.go:503: canonical request 0c72eedcb45a4bbeb1ea39238b22e711: GET\n/\n\nhost:object.51vr.local\nuser-agent:aws-sdk-dotnet-45/3.5.0.3 aws-sdk-dotnet-core/3.5.1.1 .NET_Runtime/4.0 .NET_Framework/4.0 OS/Microsoft_Windows_NT_6.2.9200.0 ClientAsync\nx-amz-content-sha256:e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855\nx-amz-date:20200901T113721Z\n\nhost;user-agent;x-amz-content-sha256;x-amz-date\ne3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
2020/09/01 19:37:21.096581 [DEBUG] policy.go:212: policyCheck: no bucket specified: requestID(0c72eedcb45a4bbeb1ea39238b22e711)
2020/09/01 19:37:21.096602 [DEBUG] client.go:196: httpRequest: merge request url: method(GET) url(http://10.2.10.62:17010/user/akInfo?ak=eJdpNVBf6dS3psFz) bodyLength[0].
2020/09/01 19:37:21.098149 [DEBUG] api_middleware.go:124: traceMiddleware: action(ListBuckets) requestID(0c72eedcb45a4bbeb1ea39238b22e711) host(object.51vr.local) method(GET) url(/) header({User-Agent:[aws-sdk-dotnet-45/3.5.0.3 aws-sdk-dotnet-core/3.5.1.1 .NET_Runtime/4.0 .NET_Framework/4.0 OS/Microsoft_Windows_NT_6.2.9200.0 ClientAsync],X-Amz-Date:[20200901T113721Z],X-Amz-Content-Sha256:[e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855],Authorization:[AWS4-HMAC-SHA256 Credential=eJdpNVBf6dS3psFz/20200901/us-east-1/s3/aws4_request, SignedHeaders=host;user-agent;x-amz-content-sha256;x-amz-date, Signature=7dd30e4b74fb7d5a67bd9a7217c440fd47d187cba1353b530c5f5be7e3ccad77],Connection:[Keep-Alive]}) remote(10.2.10.37) cost(3.509716ms)
2020/09/01 19:37:21.149837 [DEBUG] fs_manager.go:89: syncVolumeInit: get volume init lock: volume(test1)
2020/09/01 19:37:21.149925 [DEBUG] client.go:196: httpRequest: merge request url: method(GET) url(http://10.2.10.62:17010/admin/getIp) bodyLength[0].
2020/09/01 19:37:21.151707 [DEBUG] client.go:196: httpRequest: merge request url: method(GET) url(http://10.2.10.62:17010/client/volStat?name=test1) bodyLength[0].
2020/09/01 19:37:21.153603 [DEBUG] fs_manager.go:93: syncVolumeInit: release volume init lock: volume(test1)
2020/09/01 19:37:21.153880 [DEBUG] api_middleware.go:124: traceMiddleware: action(CreateBucket) requestID(c356cc9d160542e681095ea8cb760e8f) host(test1.object.51vr.local) method(PUT) url(/) header({User-Agent:[aws-sdk-dotnet-45/3.5.0.3 aws-sdk-dotnet-core/3.5.1.1 .NET_Runtime/4.0 .NET_Framework/4.0 OS/Microsoft_Windows_NT_6.2.9200.0 ClientAsync],X-Amz-Date:[20200901T113721Z],Content-Length:[0],Connection:[Keep-Alive],Content-Type:[application/xml],Content-Md5:[1B2M2Y8AsgTpgAmY7PhCfg==],X-Amz-Content-Sha256:[e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855],Authorization:[AWS4-HMAC-SHA256 Credential=eJdpNVBf6dS3psFz/20200901/us-east-1/s3/aws4_request, SignedHeaders=content-md5;content-type;host;user-agent;x-amz-content-sha256;x-amz-date, Signature=96540808ff413ead1d6832fc7982f228e6259d853839c44dc9d27162b3326d11]}) remote(10.2.10.37) cost(137.296µs)
2020/09/01 19:37:55.149568 [DEBUG] client.go:196: httpRequest: merge request url: method(GET) url(http://10.2.10.62:17010/client/partitions?name=test) bodyLength[0].
2020/09/01 19:37:55.151981 [DEBUG] client.go:196: httpRequest: merge request url: method(GET) url(http://10.2.10.62:17010/admin/getCluster) bodyLength[0].
2020/09/01 19:38:55.149554 [DEBUG] client.go:196: httpRequest: merge request url: method(GET) url(http://10.2.10.62:17010/client/partitions?name=test) bodyLength[0].
2020/09/01 19:38:55.151620 [DEBUG] client.go:196: httpRequest: merge request url: method(GET) url(http://10.2.10.62:17010/admin/getCluster) bodyLength[0].
2020/09/01 19:39:54.313018 [DEBUG] client.go:196: httpRequest: merge request url: method(GET) url(http://10.2.10.62:17010/user/akInfo?ak=eJdpNVBf6dS3psFz) bodyLength[0].
2020/09/01 19:39:54.314286 [DEBUG] auth_signature_v4.go:503: canonical request fd424b957e774754a1b6545adf9a1fee: GET\n/\n\nhost:object.51vr.local\nuser-agent:aws-sdk-dotnet-45/3.5.0.3 aws-sdk-dotnet-core/3.5.1.1 .NET_Runtime/4.0 .NET_Framework/4.0 OS/Microsoft_Windows_NT_6.2.9200.0 ClientAsync\nx-amz-content-sha256:e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855\nx-amz-date:20200901T113954Z\n\nhost;user-agent;x-amz-content-sha256;x-amz-date\ne3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
2020/09/01 19:39:54.314323 [DEBUG] policy.go:212: policyCheck: no bucket specified: requestID(fd424b957e774754a1b6545adf9a1fee)
2020/09/01 19:39:54.314335 [DEBUG] client.go:196: httpRequest: merge request url: method(GET) url(http://10.2.10.62:17010/user/akInfo?ak=eJdpNVBf6dS3psFz) bodyLength[0].
2020/09/01 19:39:54.315282 [DEBUG] api_middleware.go:124: traceMiddleware: action(ListBuckets) requestID(fd424b957e774754a1b6545adf9a1fee) host(object.51vr.local) method(GET) url(/) header({User-Agent:[aws-sdk-dotnet-45/3.5.0.3 aws-sdk-dotnet-core/3.5.1.1 .NET_Runtime/4.0 .NET_Framework/4.0 OS/Microsoft_Windows_NT_6.2.9200.0 ClientAsync],X-Amz-Date:[20200901T113954Z],X-Amz-Content-Sha256:[e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855],Authorization:[AWS4-HMAC-SHA256 Credential=eJdpNVBf6dS3psFz/20200901/us-east-1/s3/aws4_request, SignedHeaders=host;user-agent;x-amz-content-sha256;x-amz-date, Signature=ac4f60be44e6df4bc6199dce13b134b102b32ac7729a597fad52a7148584fd35],Connection:[Keep-Alive]}) remote(10.2.10.37) cost(2.303444ms)
2020/09/01 19:39:54.371105 [DEBUG] fs_manager.go:89: syncVolumeInit: get volume init lock: volume(test1)
2020/09/01 19:39:54.371137 [DEBUG] client.go:196: httpRequest: merge request url: method(GET) url(http://10.2.10.62:17010/admin/getIp) bodyLength[0].
2020/09/01 19:39:54.372572 [DEBUG] client.go:196: httpRequest: merge request url: method(GET) url(http://10.2.10.62:17010/client/volStat?name=test1) bodyLength[0].
2020/09/01 19:39:54.373871 [DEBUG] fs_manager.go:93: syncVolumeInit: release volume init lock: volume(test1)
2020/09/01 19:39:54.374039 [DEBUG] api_middleware.go:124: traceMiddleware: action(CreateBucket) requestID(ea69215bfc1d440e98b7ff57cd184e28) host(test1.object.51vr.local) method(PUT) url(/) header({User-Agent:[aws-sdk-dotnet-45/3.5.0.3 aws-sdk-dotnet-core/3.5.1.1 .NET_Runtime/4.0 .NET_Framework/4.0 OS/Microsoft_Windows_NT_6.2.9200.0 ClientAsync],X-Amz-Content-Sha256:[e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855],Authorization:[AWS4-HMAC-SHA256 Credential=eJdpNVBf6dS3psFz/20200901/us-east-1/s3/aws4_request, SignedHeaders=content-md5;content-type;host;user-agent;x-amz-content-sha256;x-amz-date, Signature=51b5cad0f4e82edaf5c9fa5f6430b37d66cee2fc2191859992b159f8d4589017],Content-Type:[application/xml],X-Amz-Date:[20200901T113954Z],Content-Length:[0],Connection:[Keep-Alive],Content-Md5:[1B2M2Y8AsgTpgAmY7PhCfg==]}) remote(10.2.10.37) cost(104.598µs)
2020/09/01 19:39:55.149562 [DEBUG] client.go:196: httpRequest: merge request url: method(GET) url(http://10.2.10.62:17010/client/partitions?name=test) bodyLength[0].
2020/09/01 19:39:55.151500 [DEBUG] client.go:196: httpRequest: merge request url: method(GET) url(http://10.2.10.62:17010/admin/getCluster) bodyLength[0].
2020/09/01 19:39:55.157836 [DEBUG] client.go:196: httpRequest: merge request url: method(POST) url(http://10.2.10.62:17010/client/vol?name=test) bodyLength[0].
2020/09/01 19:39:55.159298 [DEBUG] client.go:196: httpRequest: merge request url: method(GET) url(http://10.2.10.62:17010/client/volStat?name=test) bodyLength[0].
2020/09/01 19:40:55.149666 [DEBUG] client.go:196: httpRequest: merge request url: method(GET) url(http://10.2.10.62:17010/client/partitions?name=test) bodyLength[0].
2020/09/01 19:40:55.151507 [DEBUG] client.go:196: httpRequest: merge request url: method(GET) url(http://10.2.10.62:17010/admin/getCluster) bodyLength[0].
2020/09/01 19:41:55.149545 [DEBUG] client.go:196: httpRequest: merge request url: method(GET) url(http://10.2.10.62:17010/client/partitions?name=test) bodyLength[0].
2020/09/01 19:41:55.151267 [DEBUG] client.go:196: httpRequest: merge request url: method(GET) url(http://10.2.10.62:17010/admin/getCluster) bodyLength[0].
2020/09/01 19:42:55.149594 [DEBUG] client.go:196: httpRequest: merge request url: method(GET) url(http://10.2.10.62:17010/client/partitions?name=test) bodyLength[0].
2020/09/01 19:42:55.155080 [DEBUG] client.go:196: httpRequest: merge request url: method(GET) url(http://10.2.10.62:17010/admin/getCluster) bodyLength[0].
2020/09/01 19:43:55.149509 [DEBUG] client.go:196: httpRequest: merge request url: method(GET) url(http://10.2.10.62:17010/client/partitions?name=test) bodyLength[0].
2020/09/01 19:43:55.150953 [DEBUG] client.go:196: httpRequest: merge request url: method(GET) url(http://10.2.10.62:17010/admin/getCluster) bodyLength[0].
2020/09/01 19:44:55.149596 [DEBUG] client.go:196: httpRequest: merge request url: method(GET) url(http://10.2.10.62:17010/client/partitions?name=test) bodyLength[0].
2020/09/01 19:44:55.150907 [DEBUG] client.go:196: httpRequest: merge request url: method(GET) url(http://10.2.10.62:17010/admin/getCluster) bodyLength[0].
2020/09/01 19:44:55.161135 [DEBUG] client.go:196: httpRequest: merge request url: method(POST) url(http://10.2.10.62:17010/client/vol?name=test) bodyLength[0].
2020/09/01 19:44:55.162435 [DEBUG] client.go:196: httpRequest: merge request url: method(GET) url(http://10.2.10.62:17010/client/volStat?name=test) bodyLength[0].
```

**Comment by mervinkid** [comment]
Hi @wsj0000 

Thanks for your feedback.

It has been confirmed that this is a bug. 
Some compatibility logic in signature verification process has problems when processing CreateBucket requests.

We will repair it immediately.

**Comment by mervinkid** [comment]
This issue have been fixed by pull request #870 which have been merged into `master` branch.

**Comment by wsj0000** [comment]
> This issue have been fixed by pull request #870 which have been merged into `master` branch.

Thanks indeed！
